lib/pacemaker/pcmk_sched_location.c

```diff
@@ -135,33 +135,43 @@ score_from_rule(const xmlNode *rule_xml, int *score)
  * \internal
  * \brief Get a rule score from a node attribute
  *
- * \param[in] rule       Rule ID (for logging only)
- * \param[in] attr_name  Name of node attribute with score
- * \param[in] node       Node to get attribute for
- * \param[in] rsc        Resource being located
+ * \param[in]  constraint_id  Location constraint ID (for logging only)
+ * \param[in]  attr_name      Name of node attribute with score
+ * \param[in]  node           Node to get attribute for
+ * \param[in]  rsc            Resource being located
+ * \param[out] score          Where to store parsed score
  *
- * \return Score parsed from value of \p attr_name on \p node (default 0)
+ * \return Standard Pacemaker return code (pcmk_rc_ok if a valid score was
+ *         parsed, ENXIO if the node attribute was unset, and some other value
+ *         if the node attribute value was invalid)
  */
 static int
-score_from_attr(const char *rule, const char *attr_name,
-                const pcmk_node_t *node, const pcmk_resource_t *rsc)
+score_from_attr(const char *constraint_id, const char *attr_name,
+                const pcmk_node_t *node, const pcmk_resource_t *rsc, int *score)
 {
-    int score = 0;
     int rc = pcmk_rc_ok;
     const char *target = NULL;
     const char *score_s = NULL;
 
     target = g_hash_table_lookup(rsc->priv->meta,
                                  PCMK_META_CONTAINER_ATTRIBUTE_TARGET);
     score_s = pcmk__node_attr(node, attr_name, target, pcmk__rsc_node_current);
+    if (pcmk__str_empty(score_s)) {
+        crm_info("Ignoring location %s for %s on %s "
+                 "because it has no node attribute %s",
```

Comment:
slightly misleading for empty string value but it's close enough and the user should be smart enough to figure it out